Source/Particles/PhysicalParticleContainer.cpp

```diff
@@ -1737,6 +1833,25 @@ PhysicalParticleContainer::AddPlasmaFlux (PlasmaInjector const& plasma_injector,
                     }
                 }
                 const Real weight = t_weight;
+#elif defined(WARPX_DIM_RSPHERE)
+                // The particle weight is proportional to the user-specified
+                // flux and the emission surface within
+                // one cell (captured partially by `scale_fac`).
+                // For spherical emission (flux_normal_axis==0),
+                // the emission surface depends on
+                // the radius ; thus, the calculation is finalized here
+                Real t_weight = flux * scale_fac * dt;
+                if (loc_flux_normal_axis == 0) {
+                    if (radially_weighted) {
+                         t_weight *= 4._rt/3._rt*MathConst::pi*radial_position*radial_position;
+                    } else {
+                         // This is not correct since it might shift the particle
+                         // out of the local grid
+                         ppos.x = std::cbrt(radial_position*rmax);
+                         t_weight *= dx[0];
```

Comment:
Would there be a "correct" implementation instead?

Reply:
This is implemented in PR #5009.